src/code/z_vismono.c

```diff
@@ -22,47 +39,89 @@ void VisMono_Init(VisMono* this) {
 }
 
 void VisMono_Destroy(VisMono* this) {
-    SystemArena_FreeDebug(this->monoDl, "../z_vismono.c", 137);
+    SystemArena_FreeDebug(this->dList, "../z_vismono.c", 137);
 }
 
-void VisMono_UpdateTexture(VisMono* this, u16* tex) {
+void VisMono_UpdateTLUT(VisMono* this, u16* tlut) {
     s32 i;
 
     for (i = 0; i < 256; i++) {
-        tex[i] = ((((i >> 3 & 0x1F) * 2 + (i << 2 & 0x1F) * 4) * 0xFF / 0xD9) << 8) |
-                 (((i >> 6 & 0x1F) * 4 + (i >> 1 & 0x1F)) * 0xFF / 0xD9);
+        // `tlut[i]` is a IA16 color
+        // `i` corresponds to either byte of a RGBA16 color RRRR_RGGG GGBB_BBBA from the color frame buffer
+
+        // The high byte I (intensity) corresponds to `i` being interpreted as the high byte RRRR_RGGG
+        // I = (RRRRR * FAC_RED + GGG00 * FAC_GREEN) * (255 / FAC_NORM)
+
+        // The low byte A (alpha) corresponds to `i` being interpreted as the low byte GGBB_BBBA
+        // A = (000GG * FAC_GREEN + BBBBB * FAC_BLUE) * (255 / FAC_NORM)
+
+        // Note: I + A = (RRRRR * FAC_RED + GGGGG * FAC_GREEN + BBBBB * FAC_BLUE) * (255 / FAC_NORM)
+
+        tlut[i] = GPACK_IA16(
+            ((i >> 3 & 0x1F) * VISMONO_FAC_RED + (i << 2 & 0x1F) * VISMONO_FAC_GREEN) * 255 / VISMONO_FAC_NORM,
+            ((i >> 6 & 0x1F) * VISMONO_FAC_GREEN + (i >> 1 & 0x1F) * VISMONO_FAC_BLUE) * 255 / VISMONO_FAC_NORM);
     }
 }
 
-Gfx* VisMono_DrawTexture(VisMono* this, Gfx* gfx) {
+Gfx* VisMono_UpdateDList(VisMono* this, Gfx* gfx) {
```

Comment:
"updating a dlist" doesnt really make much sense to me, as its something that gets generated and populated every frame. is there possibly a better way to convey this meaning?

Reply:
right, it's called from draw functions so "update" is pretty bad

I guess the idea is the same as the `EnSth_AllocColorDList`-like functions except `VisMono_DrawTexture` writes to a buffer instead of allocating the buffer itself

with going the naming convention we loosely follow for these maybe `VisMono_DesaturateDList`?

and also `VisMono_UpdateTLUT` -> `VisMono_DesaturateTLUT` I guess

or MakeDList / MakeTLUT idk

Reply:
I like that desaturate tells me what its doing without even having to read its contents. would work fine with me